### PR TITLE
feat override synthesizer options from config

### PR DESCRIPTION
**Pull request by xquanluu**
https://github.com/jambonz/jambonz-feature-server/issues/589

### CODE REVIEW

**Comment by xquanluu** [comment]
Tested on jambonz.me

File: lib/tasks/say.js [hunk 1]
```diff
@@ -68,6 +68,12 @@ class TaskSay extends Task {
     const salt = cs.callSid;
 
     let credentials = cs.getSpeechCredentials(vendor, 'tts', label);
+
+    // override Synthesizer options from config verb
+    if (cs.synthesizer && cs.synthesizer.options) {
```
Comment (Catharsis68) [inline lib/tasks/say.js:73]: @xquanluu  I would add these conditions: 
- call session synthesizer options are only applied, when there is no say.synthesizer.options
- at least one "key" must be present in call session synthesizer options 
```
if (Object.keys(this.options).length === 0 && cs.synthesizer && cs.synthesizer.options && Object.keys(cs.synthesizer.options).length > 0) {
```

Comment (xquanluu) [reply lib/tasks/say.js:73]: Thanks @Catharsis68 for the review:
1/ call session synthesizer options are only applied, when there is no say.synthesizer.options
--> This seem to me we violate the rule for config verb to override configuration from other verb.

2/ at least one "key" must be present in call session synthesizer options
--> Do you know any case that user use config verb to put {} empty configuration, IMHO I think there is 2 normal case there is  synthesizer.options and there is no synthesizer.options?

Comment (davehorton) [reply lib/tasks/say.js:73]: I agree with @Catharsis68 on this one -- the config provides session-level settings, but if a verb specifies something different, that should take preference for that verb.  Given that, I think we can merge this @xquanluu ?

File: lib/tasks/say.js [hunk 2]
```diff
@@ -70,7 +70,10 @@ class TaskSay extends Task {
     let credentials = cs.getSpeechCredentials(vendor, 'tts', label);
 
     // override Synthesizer options from config verb
-    if (cs.synthesizer && cs.synthesizer.options) {
+    if (!Object.keys(this.options).length != 0 &&
```
[the remark this answers is not in the document]
Comment (davehorton) [reply lib/tasks/say.js:79]: if not length does not equal zero.....can't that be made into a simple test: if length equals zero?

**Review by davehorton — CHANGES_REQUESTED** [review]

**Pull request closed without merging**